src/unix/linux_like/linux/mod.rs

```diff
@@ -1213,6 +1213,100 @@ s! {
     }
 
     // linux/if_xdp.h
+
+    pub struct sockaddr_xdp {
+        pub sxdp_family: crate::__u16,
+        pub sxdp_flags: crate::__u16,
+        pub sxdp_ifindex: crate::__u32,
+        pub sxdp_queue_id: crate::__u32,
+        pub sxdp_shared_umem_fd: crate::__u32,
+    }
+
+    pub struct xdp_ring_offset {
+        pub producer: crate::__u64,
+        pub consumer: crate::__u64,
+        pub desc: crate::__u64,
+        pub flags: crate::__u64,
+    }
+
+    pub struct xdp_mmap_offsets {
+        pub rx: xdp_ring_offset,
+        pub tx: xdp_ring_offset,
+        pub fr: xdp_ring_offset,
+        pub cr: xdp_ring_offset,
+    }
+
+    #[deprecated(
+        since = "0.2.167",
+        note = "We consider removing this as the newer version of this struct will work with older kernel versions. If you're using it, please comment on https://github.com/rust-lang/libc/issues/4168"
```

Comment:
Slight rewording to suggest an alternative, and to rewrap to 100 characters. Could you update the others as well?

```suggestion
        note = "Use `xdp_ring_offset` instead, which will still work with old kernel versions. If \
                you still need this version, please comment at \
                https://github.com/rust-lang/libc/issues/4168"
```

After this please squash and we should be able to merge.